src/hotspot/share/compiler/compileBroker.hpp

```diff
@@ -296,15 +296,15 @@ class CompileBroker: AllStatic {
   static void compilation_init_phase1(JavaThread* THREAD);
   static void compilation_init_phase2();
   static void init_compiler_thread_log();
-  static nmethod* compile_method(const methodHandle& method,
+  static CodeBlob* compile_method(const methodHandle& method,
                                  int osr_bci,
                                  int comp_level,
                                  const methodHandle& hot_method,
                                  int hot_count,
                                  CompileTask::CompileReason compile_reason,
                                  TRAPS);
 
-  static nmethod* compile_method(const methodHandle& method,
+  static CodeBlob* compile_method(const methodHandle& method,
```

Comment:
Not so sure about these changes. It seems to me that if a method is requested to be compiled, it should always result in an nmethod.

Alternatively, would it be possible to keep these functions returning an `nmethod` but add an assert at the start to check that the passed `method` is not a method handle intrinsic?